pkg/skuba/actions/cluster/init/init.go

```diff
@@ -66,6 +78,59 @@ func (initConfiguration InitConfiguration) ControlPlaneHostAndPort() string {
 	return fmt.Sprintf("%s:%s", controlPlaneHost, controlPlanePort)
 }
 
+func (initConfiguration InitConfiguration) APIServer() KubernetesComponent {
+	return APIServer
+}
+
+func (initConfiguration InitConfiguration) ControllerManager() KubernetesComponent {
+	return ControllerManager
+}
+
+func (initConfiguration InitConfiguration) Kubelet() KubernetesComponent {
+	return Kubelet
+}
+
+func (initConfiguration InitConfiguration) ComponentExtraArgs(kubernetesComponent KubernetesComponent) map[string]string {
+	extraArgs := map[string]string{}
+	// APIServer, ControllerManager and Kubelet needs `cloud-provider` arg set if CPI is enabled
+	if initConfiguration.CloudProvider != "" {
+		switch kubernetesComponent {
+		case APIServer, ControllerManager, Kubelet:
+			extraArgs["cloud-provider"] = initConfiguration.CloudProvider
+		}
+	}
+	switch initConfiguration.CloudProvider {
+	case "aws":
+		if kubernetesComponent == ControllerManager {
+			extraArgs["allocate-node-cidrs"] = "false"
+		}
+	case "openstack":
+		switch kubernetesComponent {
+		case APIServer, ControllerManager, Kubelet:
+			extraArgs["cloud-config"] = skuba.OpenstackConfigRuntimeFile()
+		}
+	}
```

Comment:
I would prefer cloud provider first, Kubernetes component later. [1]

This makes the logic more clear.

```
// APIServer, ControllerManager and Kubelet needs `cloud-provider` arg set if CPI is enabled
switch initConfiguration.CloudProvider {
case "aws":
	switch kubernetesComponent {
	case APIServer, Kubelet:
		extraArgs["cloud-provider"] = initConfiguration.CloudProvider
	case ControllerManager:
		extraArgs["cloud-provider"] = initConfiguration.CloudProvider
		extraArgs["allocate-node-cidrs"] = "false"
	}
case "openstack":
	switch kubernetesComponent {
	case APIServer, ControllerManager, Kubelet:
		extraArgs["cloud-provider"] = initConfiguration.CloudProvider
		extraArgs["cloud-config"] = skuba.OpenstackConfigRuntimeFile()
	}
}
```

Reply:
I will update, I have no real preference, but that duplication of `cloud-provider` seems odd :)